nexus/tests/integration_tests/metrics.rs

```diff
@@ -18,17 +21,26 @@ pub async fn query_for_metrics(
     measurements
 }
 
-pub async fn query_for_latest_metric(
-    client: &ClientTestContext,
-    path: &str,
+pub async fn get_latest_system_metric(
+    cptestctx: &ControlPlaneTestContext<omicron_nexus::Server>,
+    metric_name: &str,
+    resource_id: Uuid,
 ) -> i64 {
+    let client = &cptestctx.external_client;
+    let end_time = Utc::now();
+    let url = format!(
+        "/v1/system/metrics/{metric_name}?start_time={:?}&end_time={:?}&id={:?}&order=descending", 
```

Comment:
When I add `limit=1` to this query string, I get this thrilling 400:

```
unable to parse query string: invalid type: string \"1\", expected a nonzero u32","error_message_internal":"unable to parse query string: invalid type: string \"1\", expected a nonzero u3
```

Now clearly `limit` is on there, because it's attempting to parse it, and it knows it's supposed to be a number, but something very stupid is happening. I have a hunch it's `flatten` breaking everything as usual.

https://github.com/oxidecomputer/omicron/blob/0d8552d8e46c7a9f8e55ba309747e4a308c36f5a/nexus/src/external_api/http_entrypoints.rs#L4294-L4305

@zephraph @ahl help?

Reply:
> unable to parse query string: invalid type: string \"1\", expected a nonzero u32","error_message_internal":"unable to parse query string: invalid type: string \"1\", expected a nonzero u3

Hey, it's my least favorite error ever. This is a result of `serde(flatten)`ing the pagination params. We had the same problem in MGS, and put some notes and a workaround here: https://github.com/oxidecomputer/omicron/blob/cf3a2caede0fb8f501a31b33ae3d7b54a12409aa/gateway/src/http_entrypoints.rs#L434-L464

but that required changing the type-being-flattened, which in this case is defined by dropshot. I don't know if we want to do the same kind of thing in dropshot itself, or figure out a similar-but-not-the-same workaround here?

Reply:
After a moderate amount of suffering due to trying to get this to work in Dropshot (limit is optional, which makes it a bit harder), I think I might be able to do it by — get this — _not_ flattening `PaginationParams`.

Reply:
Ah yes, here it is: https://github.com/oxidecomputer/omicron/pull/2772/files#diff-aee01ef600c8eb547421397e5d89c41a67dd6e6ca6033676cac47b7909e1e344R131-R162

Reply:
Ugh. I thought I could invert the structure like this

<img width="1295" alt="image" src="https://github.com/oxidecomputer/omicron/assets/3612203/3fab4030-9de0-45cf-a195-f8c2b9f3c56a">

but when the stuff is "inside" `PaginationParams` it becomes kind of annoying to access because it's inside this `WhichPage` enum, which you have to match on and pull things out of.

https://github.com/oxidecomputer/dropshot/blob/4be76343647319bd95efd63b081670c872afd427/dropshot/src/pagination.rs#L206-L227

Reply:
If I recall correctly, at one time one way around this was to accept two different `Query` extractors in your endpoint function.  I don't remember if this still works.  @ahl do you remember where we landed on this?

Reply:
That worked, did it in 8bbae042fff6b8ce4638b9a311ae2de3f29f777e. I tried the other way with the special deserializer and it was hard to get working with the Option. (John's example was a non-optional param.) I'm sure it can be done but the two args trick is much easier and doesn't require a change in Dropshot.